feat: move router options to own key (#5985)
fastify/fastify: docs/Reference/Server.md

  - [`forceCloseConnections`](#forcecloseconnections)
  - [`maxRequestsPerSocket`](#maxrequestspersocket)
  - [`requestTimeout`](#requesttimeout)
  - [`ignoreDuplicateSlashes`](#ignoreduplicateslashes)
  - [`bodyLimit`](#bodylimit)
  - [`onProtoPoisoning`](#onprotopoisoning)
  - [`onConstructorPoisoning`](#onconstructorpoisoning)
  - [`logger`](#logger)
  - [`loggerInstance`](#loggerInstance)
  - [`disableRequestLogging`](#disablerequestlogging)
  - [`serverFactory`](#serverfactory)
  - [`requestIdHeader`](#requestidheader)
  - [`requestIdLogLabel`](#requestidloglabel)
  - [`genReqId`](#genreqid)
  - [`trustProxy`](#trustproxy)
  - [`pluginTimeout`](#plugintimeout)
  - [`exposeHeadRoutes`](#exposeheadroutes)
  - [`return503OnClosing`](#return503onclosing)
  - [`ajv`](#ajv)
  - [`serializerOpts`](#serializeropts)
  - [`http2SessionTimeout`](#http2sessiontimeout)
  - [`frameworkErrors`](#frameworkerrors)
  - [`clientErrorHandler`](#clienterrorhandler)
  - [`rewriteUrl`](#rewriteurl)

>   - [RouterOptions](#routeroptions)

Reviewer: we should move `ignoreDuplicateSlashes` too

Author: Good catch should be updated now

}
```

## RouterOptions
<a id="routeroptions"></a>

> Fastify uses [find-my-way](https://github.com/delvedor/find-my-way) for its
> HTTP router. With routerOptions you can pass 
> [find-my-way options](https://github.com/delvedor/find-my-way?tab=readme-ov-file#findmywayoptions)
> to customize your HTTP router within Fastify.

Reviewer: ```suggestion
Fastify uses [`find-my-way`](https://github.com/delvedor/find-my-way) for its
HTTP router. The `routerOptions` parameter allows passing 
[`find-my-way` options](https://github.com/delvedor/find-my-way?tab=readme-ov-file#findmywayoptions)
to customize the HTTP router within Fastify.
```

Author: Should be fixed now
